javaparser-core/src/main/java/com/github/javaparser/HasParentNode.java

```diff
@@ -51,15 +51,28 @@
 
     /**
      * Get the ancestor of the node having the given type, or null if no ancestor of the given type is found.
+     *
+     * @deprecated use findParent
      */
+    @Deprecated
     default <N> Optional<N> getAncestorOfType(Class<N> classType) {
-        Node parent = getParentNode().orElse(null);
-        while (parent != null) {
-            if (classType.isAssignableFrom(parent.getClass())) {
-                return Optional.of(classType.cast(parent));
+        return findParent(classType);
+    }
+
+    /**
+     * Walks the parents of this node, returning the first node of type "type" or empty() if none is found.
+     * Also works for other types, like the With... interfaces.
+     */
+    default <N> Optional<N> findParent(Class<N> type) {
```

Comment:
would be "ancestor" more correct here? The parent is only the immediate containing node, right?

Reply:
Yeah, I guess we can do that.

Reply:
I agree, "findAncestor()" would probably be the best name. :-)

Except that this gives us *two* deprecated methods now, "findParent()" and "getAncestorOfType()", but, c'est la vie!